runtime/lua/vim/ui.lua

```diff
@@ -104,4 +104,66 @@ function M.input(opts, on_confirm)
   end
 end
 
+--- Opens a URI by sending it to the system's default handler.
+--- This function utilizes `xdg-open`, `wslview`, or `open` commands depending on
+--- the operating system to open the provided URI.
+---
+---
+---@param uri string Uri to be opened
+---
+---@param opts table|nil Optional keyword arguments:
+---             - confirm boolean|nil Ask user for confirmation before opening.
+---
+---@return string|nil returns the string output of the command. nil if
+---opts.confirm and the user declined. See |system()|.
+---
+--- Example:
+--- <pre>lua
+--- vim.ui.open_uri("https://neovim.io/")
+---
+--- vim.ui.open_uri("/path/to/file")
+--- </pre>
+function M.open_uri(uri, opts)
+  vim.validate({
+    opts = { opts, 'table', true },
+  })
+
+  opts = opts or {}
+  local cmd
+  local sysname = vim.loop.os_uname().sysname
+
+  if sysname == 'Darwin' then
+    cmd = 'open'
+  elseif sysname:match('windows') then
+    cmd = 'wslview'
```

Comment:
Is `wslview` always available on windows?

Reply:
Is there a good way to detect wsl or default windows cmd? Otherwise on the windows host, `start` also works I think.

Reply:
Apparently [`/proc/sys/fs/binfmt_misc/WSLInterop`](https://superuser.com/questions/1749781/how-can-i-check-if-the-environment-is-wsl-from-a-shell-script#:~:text=The%20presence%20of,WSL%20at%20startup.) is a good indicator